Makefile

```diff
@@ -0,0 +1,109 @@
+# Run `make help` to display help
+
+# --- Global -------------------------------------------------------------------
+O = out
+COVERAGE = 15
+VERSION ?= $(shell git describe --tags --dirty  --always)
+REPO_ROOT = $(shell git rev-parse --show-toplevel)
+
+all: build test check-coverage lint ## build, test, check coverage and lint
+	@if [ -e .git/rebase-merge ]; then git --no-pager log -1 --pretty='%h %s'; fi
+	@echo '$(COLOUR_GREEN)Success$(COLOUR_NORMAL)'
+
+ci: clean check-uptodate all  ## Full clean build and up-to-date checks as run on CI
+
+check-uptodate: tidy
+	test -z "$$(git status --porcelain -- go.mod go.sum)" || { git status; false; }
+
+clean::  ## Remove generated files
+	-rm -rf $(O)
+
+.PHONY: all check-uptodate ci clean
+
+# --- Build --------------------------------------------------------------------
+GO_MODULE := $(shell go list -m)
```

Comment:
this is not used so should be removed. Since it is a `:=` var, the command `go list -m` will be run every time you run `make`. If it were used, ok, but since it is not, may as well go.